micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/headers/HeadersSettingSpec.groovy

```diff
@@ -300,6 +303,32 @@ class HeadersSettingSpec extends HttpMethodSpec {
                     _ as Long)
     }
 
+    @Unroll
+    def "should have 'Authorization' header with value: '#authorizationValue'"() {
+        given:
+        httpMethodBuilder = new HttpMethodBuilder(restOperations)
+        when:
+        httpMethodBuilder
+                .get()
+                .onUrlFromTemplate(TEMPLATE_URL)
+                .withVariables(OBJECT_ID)
+                .withHeaders()
+                .basicAuthorization(username, password)
+                .andExecuteFor()
+                .anObject()
+                .ofType(BigDecimal)
+        then:
+        1 * restOperations.exchange(TEMPLATE_URL,
+                GET,
+                { HttpEntity httpEntity -> httpEntity.headers['Authorization'] == [authorizationValue] } as HttpEntity,
+                _ as Class,
+                _ as Long)
+        where:
+            username = "AuthUsername"
+            password = "AuthPassword"
+            authorizationValue = ("Basic " + encodeCredentials(username, password)) as String
```

Comment:
http://jasonrudolph.com/blog/2008/07/30/testing-anti-patterns-the-ugly-mirror/

You use the same code (`encodeCredentials()`) in both production code and assertion. If `encodeCredentials()` has a bug, assertion won't catch it. I suggest hard-coding few encoded BASIC headers, taken from some arbitrary source, e.g.:

```groovy
username  | password      || authorizationValue
'Aladdin' | 'open sesame' || 'Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=='
```

From http://en.wikipedia.org/wiki/Basic_access_authentication